tests/utils/test_serialization.py

```diff
@@ -143,6 +150,13 @@ def test_pydantic_json_compatibility() -> None:
     assert json.loads(model.json()) == json.loads(encode_json(model))
 
 
-def test_unsupported_type_raises() -> None:
-    with pytest.raises(TypeError):
-        encode_json(lambda: None)
```

Comment:
Is altering this an issue? FWICT encoders mostly raise `TypeError` but they have a test in msgspec that raises `EncodeError` if enum member type is invalid: https://github.com/jcrist/msgspec/blob/c8e316fa27b264260e28c57963c453cdf5afad85/tests/test_common.py#L315.

After this PR, the encoders will always raise `SerializationException` if either `TypeError` or `EncodeError` is raised, and there are no other test failures as a result, but is the `TypeError` relied upon somewhere given this test exists?

Reply:
This shouldn’t be an issue as such, but it might be a breaking change if users rely on a catching a `TypeError` from this.

Reply:
OK, I'll merge and if there is any blowback I'll revisit.